### PR TITLE
#163426639 Search article by title and description

### PR DESCRIPTION
**Pull request by Ndifreke**
#### What does this PR do?
- Create API to search Articles by description or title

 #### Description of Task to be completed?
- Create API at /articles/search&query{title|description}&pageLimit&pageOffset to search for an Article

 #### What are the relevant pivotal tracker stories?
- [#163426639](https://www.pivotaltracker.com/n/projects/2240406/stories/163426639)

### CODE REVIEW

File: server/tests/index.spec.js [hunk 1]
```diff
@@ -28,3 +28,4 @@ import "./models/ratings.spec";
 import "./controller/follow/follow.spec";
 import "./controller/reaction/article.spec";
 import "./controller/reaction/comment.spec";
+// import "./controller/article/searchArticle.spec";
```
Comment (benfluleck) [inline server/tests/index.spec.js]: Why is this here ?

**Review by benfluleck — CHANGES_REQUESTED** [review]

**Comment by benfluleck** [comment]
Please any reviewers test this branch locally as I have to @Ndifreke not to write any tests

**Review by eokwukwe — APPROVED** [review]

File: server/controllers/article.js [hunk 2]
```diff
@@ -311,3 +316,49 @@ export const rateArticle = async (req, res) => {
...
+  pageLimit = pageLimit < 0 ? 0 : pageLimit;
+  pageNumber = pageNumber < 0 ? 0 : pageNumber;
+  query = query || null;
+
+  // return early avoid search for empty query
+  if (!query) {
+    httpResponse(res, {
+      success: true,
+      message: "empty query supplied",
+      statusCode: 200,
```
Comment (mrkingz) [inline server/controllers/article.js]: statusCode should be 404

File: server/routes/article.js [hunk 3]
```diff
@@ -131,6 +145,17 @@ articleRouter.get(
...
+  (req, resp, next) => {
+    if ('search' in req.query) {
+      searchArticle(req, resp, next);
+    } else {
+      next();
+    }
+  }
+);
+
 /**
```
Comment (obasajujoshua31) [inline server/routes/article.js:149]: we are using res not resp.

File: server/routes/article.js [hunk 4]
```diff
@@ -60,7 +65,16 @@ articleRouter.get(
...
+articleRouter.get("/",
+  validatePaginationParameters,
+  (req, res, next) => {
+    if ('search' in req.query) {
+      next();
+    } else {
+      getAllArticles(req, res, next);
+    }
+  }
+);
```
Comment (obasajujoshua31) [inline server/routes/article.js:71]: i don't really understand what is going on here, this convention is kind of like strange to this project

**Review by ODINAKACHUKWU — APPROVED** [review]

File: server/tests/index.spec.js (hunk 1 above)
Comment (Ndifreke) [reply server/tests/index.spec.js]: I kept that in a case in the future we switch test DB, it will not be forgotten to be added to the test runner

File: server/routes/article.js (hunk 4 above)
Comment (Ndifreke) [reply server/routes/article.js:71]: Please can you make it a bit more specific what you find wrong about this

**Review by benfluleck — APPROVED** [review]
